batch/batch/cloud/gcp/driver/create_instance.py

```diff
@@ -73,6 +73,21 @@ def create_vm_config(
 
     assert instance_config.is_valid_configuration(resource_rates.keys())
 
+    def scheduling() -> dict:
+        result = {
+            'automaticRestart': False,
+            'onHostMaintenance': 'TERMINATE',
+        }
+
+        if preemptible:
+            result.update({
+                'provisioningModel': 'SPOT',
+                'instanceTerminationAction': 'DELETE',
+                'preemptible': True,
```

Comment:
Does the `preemptible` flag apply for spot model? Trying to parse a multi-level boolean logic here:
https://cloud.google.com/compute/docs/instances/create-use-spot#identify

> If the output includes the provisioningModel field set to SPOT, <...> Otherwise: <...> If the output includes the preemptible field set to true, the VM is a preemptible VM, Otherwise, the VM is a standard VM

Reply:
I've had the same question, but found this in the Pulumi docs (https://www.pulumi.com/registry/packages/gcp/api-docs/compute/instance/):

> `provisioning_model`: Describe the type of preemptible VM. This field accepts the value `STANDARD` or `SPOT`. If the value is `STANDARD`, there will be no discount. If this is set to `SPOT`, `preemptible` should be `true` and `auto_restart` should be `false`.

It also seems to work in practice -- I've seen Hail Batch VMs yesterday getting scheduled using Spot provisioning. The breakage occurred for non-preemptible workers, for which one can't specify `instanceTerminationAction`.